haskellPackages: update stackage and hackage

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #199424. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

sternenseemann committed:
haskellPackages: stackage LTS 19.31 -> LTS 19.32

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
all-cabal-hashes: 2022-11-03T21:09:38Z -> 2022-11-11T17:48:48Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskellPackages.h-raylib: remove broken flag

Closes #197407.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

natto1784 committed:
haskellPackages/extra: init brick 1.3 for swarm 0.2.0.0

Signed-off-by: Amneesh Singh <natto@weirdnatto.in>

natto1784 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

natto1784 committed:
haskellPackages.swarm: fix broken package

Signed-off-by: Amneesh Singh <natto@weirdnatto.in>

cdepillabout commented:
@natto1784 I've cherry-picked your commits from https://github.com/NixOS/nixpkgs/pull/201344 directly onto this PR.

You may also be interested in adding yourself as a maintainer so that you'll get a notification if `swarm` breaks in the future:

https://github.com/NixOS/nixpkgs/blob/3e1a3aa9392fd2be39717eb1ea2a08253d1ba3fd/pkgs/development/haskell-modules/configuration-hackage2nix/main.yaml#L166

cdepillabout committed:
haskellPackages: update ghc-lib-parser to 9.2.5.20221107 and fourmolu to 0.9.0.0

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages.language-docker_11_0_0: generate for hadolint

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann commented:
Blocker seems to be HLS (cc @maralorn) build and eval failures, cabal2nix build failure on GHC 9.2.4 is expected for the moment (arithmetic primops problem on  aarch64).

Commit:
haskell.packages.ghc94.{or,four}molu: Disable fixity-th flag

See: https://github.com/tweag/ormolu/issues/941

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages.ghc-lib: 9_2_4_20220729 -> 9_2_5_20221107

cdepillabout committed:
haskell.packages.ghc94.ghc-lib: 9_4_2_20220822 -> 9_4_3_20221104

maralorn committed:
haskellPackages.fourmolu_0_9_0_0: Fix build

maralorn committed:
Merge pull request #201692 from MercuryTechnologies/gabriella/fix_ormolu_ghc9.4

haskell.packages.ghc94.{or,four}molu: Disable `fixity-th` flag

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.monad-bayes: remove stale jailbreak

monad-bayes is now broken because it requires a too recent brick version.
Removing this jailbreak will make the nature of the problem clearer at
least.

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

cdepillabout committed:
haskellPackages: stop evaluating hyper-haskell-server-with-packages in jobset

cdepillabout committed:
haskellPackages.hsqml: disable new packages
